ubuntu/16.04/usr/local/share/container/baseimage-10.sh

```diff
@@ -25,3 +31,20 @@ check_development_start() {
 do_development_start() {
   :
 }
+
+do_build_user_ssh_keys() {
+  set +x
+  if [ -n "$BUILD_USER_SSH_PRIVATE_KEY" ] && [ -n "$BUILD_USER_SSH_PUBLIC_KEY" ]; then
+    echo "Setting up SSH keys for the build user"
+    mkdir -p /home/build/.ssh/
+    echo "$BUILD_USER_SSH_PRIVATE_KEY" | base64 --decode > /home/build/.ssh/id_rsa
+    echo "$BUILD_USER_SSH_PUBLIC_KEY" | base64 --decode > /home/build/.ssh/id_rsa.pub
+    echo "$BUILD_USER_SSH_KNOWN_HOSTS" | base64 --decode > /home/build/.ssh/known_hosts
```

Comment:
it might be a bit cumbersome to have to provide the full known_hosts file, especially with the way ubuntu requires the host/ip's to be hashed, whereas OSX doesn't automatically hash them.
It's possible to keyscan on build, however that does mean though that if MITM'd, the MITM'd key will be added (SSH doesn't by standard use any root CA trust, although it's capable of doing so)